### PR TITLE
Update ozone from 1.1.0 to 1.2.1

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?

update the ozone version from 1.1.0 to 1.2.1

### Why are the changes needed?
when update ozone from 1.1.0 to 1.2.1, some classes of ozone can not find classes of protobuf if relocation for protobuf, so ozone needs to be included for protobuf shade. Also add a shade to metrics for ratis otherwise JmxReporter.class cannot be found in the metrics-core-4.1.1.jar currently used by alluxio.

### CODE REVIEW

**Comment by maobaolong** [comment]
@Haoning-Sun Thanks for bump ozone version for alluxio, but AFAIK, Ozone community is release 1.2.1 right now, so would you like to help to bump 1.2.1?

File: core/server/common/src/test/java/alluxio/master/journal/raft/SnapshotReplicationManagerTest.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public void before() throws Exception {
 
   private SimpleStateMachineStorage getSimpleStateMachineStorage() throws IOException {
     RaftStorage rs = new RaftStorageImpl(mFolder.newFolder(CommonUtils.randomAlphaNumString(6)),
-            RaftServerConfigKeys.Log.CorruptionPolicy.getDefault());
+            RaftServerConfigKeys.Log.CorruptionPolicy.getDefault(), 0);
```
Comment (maobaolong) [inline core/server/common/src/test/java/alluxio/master/journal/raft/SnapshotReplicationManagerTest.java:123]: Are these changes have related with this PR?

**Review by maobaolong — CHANGES_REQUESTED** [review]
Left a comment

File: core/server/common/src/test/java/alluxio/master/journal/raft/SnapshotReplicationManagerTest.java (hunk 1 above)
Comment (Haoning-Sun) [reply core/server/common/src/test/java/alluxio/master/journal/raft/SnapshotReplicationManagerTest.java:123]: Aligned ratis version, these changes are needed.

**Comment by Haoning-Sun** [comment]
> @Haoning-Sun Thanks for bump ozone version for alluxio, but AFAIK, Ozone community is release 1.2.1 right now, so would you like to help to bump 1.2.1?

**Comment by Haoning-Sun** [comment]
> @Haoning-Sun Thanks for bump ozone version for alluxio, but AFAIK, Ozone community is release 1.2.1 right now, so would you like to help to bump 1.2.1?

I can try it if needed.

**Comment by maobaolong** [comment]
@Haoning-Sun FYI https://issues.apache.org/jira/browse/RATIS-1471 motioned, Ratis should change the non back-ward compatible changes, in the future.  Anyway, I'm fine with the change of this PR.

**Review by HelloHorizon — APPROVED** [review]
@Haoning-Sun thanks for the improvement. Overall LGTM, can you also add some descriptions about the classloader bug?

**Comment by Haoning-Sun** [comment]
when update ozone from 1.1.0 to 1.2.1, some classes of ozone can not find classes of protobuf if relocation for protobuf, so ozone needs to be included for protobuf shade. Also add a shade to metrics  for ratis otherwise JmxReporter.class cannot be found in the metrics-core-4.1.1.jar currently used by alluxio.

File: shaded/ozone/pom.xml [hunk 2]
```diff
@@ -62,15 +63,26 @@
       </activation>
       <dependencies>
         <dependency>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-ozone-filesystem</artifactId>
+          <groupId>org.apache.ozone</groupId>
+          <artifactId>ozone-filesystem</artifactId>
```
Comment (maobaolong) [inline shaded/ozone/pom.xml:67]: `hadoop-ozone-filesystem` has been changed to `ozone-filesystem-hadoop2` and `ozone-filesystem-hadoop3`

File: shaded/ozone/pom.xml [hunk 3]
```diff
@@ -62,15 +63,26 @@
...
         </dependency>
         <dependency>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-ozone-client</artifactId>
+          <groupId>org.apache.ozone</groupId>
+          <artifactId>ozone-client</artifactId>
           <version>${ufs.ozone.version}</version>
         </dependency>
+        <dependency>
+          <groupId>io.dropwizard.metrics</groupId>
```
Comment (maobaolong) [inline shaded/ozone/pom.xml:82]: Why dependent to this dependency?

File: shaded/ozone/pom.xml [hunk 4]
```diff
@@ -155,6 +171,9 @@
                   <include>org.apache.httpcomponents:*</include>
                   <include>commons-httpclient:commons-httpclient</include>
                   <include>org.codehaus.jackson:*</include>
+                  <include>org.apache.ozone:*</include>
+                  <include>org.apache.ratis:*</include>
+                  <include>io.dropwizard.metrics:metrics-core</include>
```
Comment (maobaolong) [inline shaded/ozone/pom.xml:176]: Can you explain the purpose of these three line?

File: shaded/ozone/pom.xml [hunk 5]
```diff
@@ -135,6 +147,10 @@
                   <pattern>org.codehaus.jackson</pattern>
                   <shadedPattern>${shading.prefix}.org.codehaus.jackson</shadedPattern>
                 </relocation>
+                <relocation>
```
Comment (maobaolong) [inline shaded/ozone/pom.xml:150]: What exception you encountered lead you to relocation this package?

File: pom.xml [hunk 6]
```diff
@@ -150,10 +150,10 @@
     <parquet.version>1.11.0</parquet.version>
     <protobuf.version>3.12.4</protobuf.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <ratis.version>2.0.0</ratis.version>
+    <ratis.version>2.2.0</ratis.version>
     <slf4j.version>1.7.30</slf4j.version>
     <jackson.version>2.11.1</jackson.version>
-    <ozone.version>1.1.0</ozone.version>
+    <ozone.version>1.2.1</ozone.version>
```
Comment (maobaolong) [inline pom.xml:156]: remove this and the ozone related dependencies of dependencyManagement within this file.

**Review by maobaolong — CHANGES_REQUESTED** [review]
Thanks for the improvement for Alluxio ozone ufs module, left some comments.

File: shaded/ozone/pom.xml (hunk 2 above)
Comment (maobaolong) [reply shaded/ozone/pom.xml:67]: please reference this PR https://github.com/Alluxio/alluxio/pull/14672

**Comment by maobaolong** [comment]
[Support_build_ozone_1_2_1.patch.txt](https://github.com/Alluxio/alluxio/files/7796744/Support_build_ozone_1_2_1.patch.txt)

I just tested this patch can work on my test env for ozone cluster.
